isisd/isisd.c

```diff
@@ -3331,14 +3406,122 @@ uint32_t isis_restart_read_overload_time(struct isis_area *isis_area)
 		overload_time = json_object_get_int(json_overload_time);
 	}
 
-	json_object_object_del(json_areas, area_name);
-
 	json_object_to_file_ext(filepath, json, JSON_C_TO_STRING_PRETTY);
 	json_object_free(json);
 
 	return overload_time;
 }
 
+/*
+ * Record in non-volatile memory the advertise high metrics value.
+ */
+void isis_restart_write_advertise_high_metric(struct isis_area *isis_area,
+					      uint32_t max_metric)
+{
+	char *filepath;
+	const char *area_name;
+	json_object *json;
+	json_object *json_areas;
+	json_object *json_area;
+
+	filepath = isis_restart_filepath();
+	area_name = isis_area->area_tag;
+
+	json = json_object_from_file(filepath);
+	if (json == NULL)
+		json = json_object_new_object();
+
+	json_object_object_get_ex(json, "areas", &json_areas);
+	if (!json_areas) {
+		json_areas = json_object_new_object();
+		json_object_object_add(json, "areas", json_areas);
+	}
+
+	json_object_object_get_ex(json_areas, area_name, &json_area);
+	if (!json_area) {
+		json_area = json_object_new_object();
+		json_object_object_add(json_areas, area_name, json_area);
+	}
+
+	json_object_int_add(json_area, "max_metric", max_metric);
+	json_object_to_file_ext(filepath, json, JSON_C_TO_STRING_PRETTY);
+	json_object_free(json);
+}
+
+/*
+ * Fetch from non-volatile memory the advertise max metric value.
```

Comment:
what is the purpose of storing data in non-volatile memory instead of the configuration?

Please explain in the commit log

Reply:
I'm also having a little trouble with this area: if some value is part of the config, then we need to process the config before we can use the value. we do have the mechanism needed to do that, I think?

Reply:
following up on this issue: we had some discussion during the weekly dev call this week, and it seemed like there was agreement that we don't want to store config data in this way. we have the mechanism that would allow isis to know that its config had been completely delivered, and that could trigger the sending of messages.

Reply:
Opened a separate PR to address this issue:  #12819 https://github.com/FRRouting/frr/pull/12819